posthog/database_healthcheck.py

```diff
@@ -49,13 +49,19 @@ def _get_timebucket(self) -> int:
         return round(time.time() / self.time_interval)
 
     def is_postgres_connected_check(self) -> bool:
-        try:
-            with connections[DATABASE_FOR_FLAG_MATCHING].cursor() as cursor:
-                cursor.execute("SELECT 1")
-            return True
-        except Exception:
-            logger.exception("postgres_connection_failure")
-            return False
+        for i in range(3):
+            try:
+                with connections[DATABASE_FOR_FLAG_MATCHING].cursor() as cursor:
+                    cursor.execute("SELECT 1")
+                return True
+            except Exception:
+                logger.exception(
+                    f"failed to connect to postgres {DATABASE_FOR_FLAG_MATCHING} node attempt {i + 1} of 3"
+                )
+                time.sleep(0.3)
```

Comment:
flyby that this will make some requests unacceptably slow. It's much better imo to have error computing flags fire (which doesn't mean none of the flags were evaluated, just that the ones relying on the db weren't), than to slow down the response by upto a second occasionally

Reply:
A better way to achieve the same as retries here would be to increase the query wait timeout setting, but ideally we'd tune the read replica connection settings and instances so these timeouts don't happen often. I bet when this issue happens it shows up on the pgbouncer graph as no. of waiting connections spiking (to pgbouncer, not to the db - that's been the case in the past)

Reply:
Yep that makes sense. Was thinking too much about customers who are dealing with our worst case